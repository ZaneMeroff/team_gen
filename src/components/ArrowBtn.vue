<template>

  <button
    @click="emitArrowBtnClick"
    :class="arrowDirection === 'up' || arrowDirection === 'down' ? 'arrow-btn' : 'arrow-btn nav-arrow-btn'">
    <span v-html="arrowDisplay"></span>
  </button>

</template>

<script>
  export default {
    name: "ArrowBtn",
    props: {
      arrowDirection: { type: String, default: "up" },
    },
    components: {},
    data() {
      return {
        arrowDisplay: "",
        arrowKey: {
          up:    "&#8593",
          down:  "&#8595",
          left:  "&#x2190",
          right: "&#x2192",
        },
      }
    },
    methods: {

      setArrowDisplay(direction) {
        this.arrowDisplay = this.arrowKey[direction]
      },

      emitArrowBtnClick() {
        this.$emit("arrowClick")
      },
    },
    mounted() {
      this.setArrowDisplay(this.arrowDirection)
    },
    computed: {},
    watch: {},
  }
</script>

<style scoped>

  .arrow-btn {
    height: 75px;
    width: 75px;
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 3rem;
  }

  .nav-arrow-btn {
    border-radius: 37.5px;
    padding-bottom: 7px;
  }

  .arrow-btn:hover {
    cursor: pointer;
  }
</style>
