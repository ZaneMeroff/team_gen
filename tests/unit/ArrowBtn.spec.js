import ArrowBtn from "@/components/ArrowBtn"
import { shallowMount } from "@vue/test-utils"

describe("ArrowBtn", () => {

  describe("snapshots", () => {

    it("should render with default html", () => {
      const component = shallowMount(ArrowBtn)

      expect(component.element).toMatchSnapshot()
    })

    it("should render with nav-arrow-btn class", () => {
      const propsData = { arrowDirection: "left" }
      const component = shallowMount(ArrowBtn, { propsData })

      expect(component.element).toMatchSnapshot()
    })
  })

  describe("props", () => {

    describe("arrowDirection", () => {

      it("should have a default value", () => {
        const component = shallowMount(ArrowBtn)

        expect(component.vm.arrowDirection).toEqual("up")
      })

      it("should accept a string", () => {
        const propsData = { arrowDirection: "down" }
        const component = shallowMount(ArrowBtn, { propsData })

        expect(component.vm.arrowDirection).toEqual(propsData.arrowDirection)
      })
    })
  })

  describe("data", () => {

    describe("arrowDisplay", () => {

      it("should return a default value", () => {
        const component = shallowMount(ArrowBtn)

        expect(component.vm.$data.arrowDisplay).toEqual("&#8593")
      })
    })

    describe("arrowKey", () => {

      it("should have a default value", () => {
        const component = shallowMount(ArrowBtn)

        const expected = {
          down:  "&#8595",
          left:  "&#x2190",
          right: "&#x2192",
          up:    "&#8593",
        }
        expect(component.vm.$data.arrowKey).toEqual(expected)
      })
    })
  })

  describe("methods", () => {

    describe("setArrowDisplay", () => {

      it("should set arrowDisplay if arrowDirection is 'up'", () => {
        const component = shallowMount(ArrowBtn)

        component.vm.setArrowDisplay("up")

        expect(component.vm.$data.arrowDisplay).toEqual("&#8593")
      })

      it("should set arrowDisplay if arrowDirection is 'down'", () => {
        const component = shallowMount(ArrowBtn)

        component.vm.setArrowDisplay("down")

        expect(component.vm.$data.arrowDisplay).toEqual("&#8595")
      })

      it("should set arrowDisplay if arrowDirection is 'left'", () => {
        const component = shallowMount(ArrowBtn)

        component.vm.setArrowDisplay("left")

        expect(component.vm.$data.arrowDisplay).toEqual("&#x2190")
      })

      it("should set arrowDisplay if arrowDirection is 'right'", () => {
        const component = shallowMount(ArrowBtn)

        component.vm.setArrowDisplay("right")

        expect(component.vm.$data.arrowDisplay).toEqual("&#x2192")
      })
    })

    describe("emitArrowBtnClick", () => {

      it("should emit arrowClick", () => {
        const component = shallowMount(ArrowBtn)

        component.vm.emitArrowBtnClick()

        const expected = { "arrowClick": [[]] }
        expect(component.emitted()).toEqual(expected)
      })
    })
  })

  describe("lifecyle hooks", () => {

    describe("mounted", () => {

      it("should call setArrowDisplay with a direction", () => {
        const spy = jest.spyOn(ArrowBtn.methods, "setArrowDisplay")
        const propsData = { arrowDirection: "down" }
        const component = shallowMount(ArrowBtn, { propsData })

        expect(spy).toHaveBeenCalledWith("down")
      })
    })
  })

  describe("user interactions", () => {

    describe("events", () => {

      it("should emit arrowClick when button is clicked", () => {
        const component = shallowMount(ArrowBtn)

        component.find(".arrow-btn").trigger("click")

        const expected = { "arrowClick": [[]] }
        expect(component.emitted()).toEqual(expected)
      })
    })
  })
})
