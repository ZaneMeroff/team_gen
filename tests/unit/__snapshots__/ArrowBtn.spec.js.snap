// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowBtn snapshots should render with default html 1`] = `
<button
  class="arrow-btn"
>
  <span />
</button>
`;
